flink-runtime/src/main/java/org/apache/flink/runtime/io/network/netty/CreditBasedClientHandler.java

```diff
@@ -37,20 +43,31 @@
 
 import java.io.IOException;
 import java.net.SocketAddress;
+import java.util.ArrayDeque;
 import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.ConcurrentMap;
 import java.util.concurrent.atomic.AtomicReference;
 
+/**
+ * Channel handler to read {@link BufferResponse} and {@link ErrorResponse} messages from the
+ * producer, to write and flush {@link AddCredit} message for the producer.
+ */
 class CreditBasedClientHandler extends ChannelInboundHandlerAdapter {
 
 	private static final Logger LOG = LoggerFactory.getLogger(CreditBasedClientHandler.class);
 
+	/** Channels, which already requested partitions from the producers. */
 	private final ConcurrentMap<InputChannelID, RemoteInputChannel> inputChannels = new ConcurrentHashMap<>();
 
+	/** Channels, which will notify the producers about unannounced credit. */
+	private final ArrayDeque<RemoteInputChannel> inputChannelsWithCredit = new ArrayDeque<>();
+
 	private final AtomicReference<Throwable> channelError = new AtomicReference<>();
 
+	private final ChannelFutureListener writeListener = new WriteAndFlushNextMessageIfPossibleListener();
+
 	/**
-	 * Set of cancelled partition requests. A request is cancelled iff an input channel is cleared
+	 * Set of cancelled partition requests. A request is cancelled if an input channel is cleared
```

Comment:
I guess, this "iff" was correct in the meaning of "if and only if" (a common abbreviation in math) - if you like to, you can replace it with the full form though if that is still the logic behind

Reply:
got it, already restored the original writing.